h2o-docs/src/product/explain.rst

```diff
@@ -0,0 +1,369 @@
+
+In Python, the ``H2OExplanation`` will always be printed, even if you save it to an object.  Once you save it to an object, however, if you want to print it again, you must do: ``from IPython.core.display import display`` and ``display(exa)``.
+
+
+
+Explain a single row prediction
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``h2o.explain_row()`` function provides model explanations for a single row of test data. Using the previous code example, you can evaluate row-level behavior by specifying the ``row_index``:
+
+.. tabs::
+   .. code-tab:: r R
+
+        # Explain row 1 with all AutoML models
+        h2o.explain_row(aml, test, row_index = 1)
+
+        # Explain row 1 with a single model
+        h2o.explain_row(aml@leader, row_index = 1)
+
+   .. code-tab:: python
+
+        # Explain row 1 with all AutoML models
+        aml.explain_row(test, row_index=1)
+
+        # Explain row 1 with a single model
+        aml.leader.explain_row(test, row_index=1)
+
+
+Output: Explanations
+--------------------
+
+TO DO: Overview of the output object.  Add some plots
+
+The output from these functions is an ``H2OExplanation``, which is essentially a list of explanations.  Most explanations are plots with some associated metadata that can be extracted.  The plots are ggplot2 objects in R and matplotlib figures in Python.  You can customize these plot objects as desired, after they are generated.  Note: You can also customize the plot objects before they are generated by utilizing the ``plot_overrides`` argument.
+
+The ``H2OExplanation`` is a named list in R and an ordered dictionary in Python, so explanations can be accessed by their name (same names as in ``include_explanations`` and ``exlude_exlpanations``).
+
+
+When ``h2o.explain()`` is provided a list of models, the following explanations will be generated by default:
+
+- Leaderboard (compare all models)
+- Confusion Matrix for Leader Model (classification only)
+- Residual Analysis for Leader Model (regression only)
+- Variable Importance of Top Base (non-Stacked) Model 
+- Variable Importance Heatmap (compare all models)
+- Model Correlation Heatmap (compare all models)
+- SHAP Summary of Top Tree-based Model (TreeSHAP)
+- PD Plots (compare all models)
+
+When ``explain()`` is provided a single model, we get the following explanations:
+
+- Variable Importance
+- Individual Conditional Expectation (ICE) Plots
+- TO DO: Finish
+
+
+
+Explanation Plotting Functions 
+------------------------------
+
+_TO DO: Let's put examples of each function and the plot, in the order in which they appear in the ``h2o.explain()`` output.  Let's also show how to customize the plots._
+
+
+
+There are a number of individual plotting functions that are used inside the ``explain()`` function.  Some of these functions 
+
+Takes a list of models (including an AutoML object or leaderboard slice) as input:
+
+.. tabs::
+   .. code-tab:: r R
+
+        h2o.varimp_plot()
+        h2o.varimp_heatmap()          
```

Comment:
inconsistencies in parentheses - some function have them here and some don't

Reply:
thx